Refactor VideoPresentationInterfaceIOS

danae404 opened the pull request:
#### 20180664c83966aecf29787875bb1f80c96623bb
<pre>
Refactor VideoPresentationInterfaceIOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=268636">https://bugs.webkit.org/show_bug.cgi?id=268636</a>
<a href="https://rdar.apple.com/122193331">rdar://122193331</a>

Reviewed by Andy Estes.

This patch extracts the AVKit-specific code from VideoPresentationInterfaceIOS
and places it into a subclass called VideoPresentationInterfaceAVKit. There
is also now a second subclass called VideoPresentationInterfaceLMK, which
will be used on the vision platform in the future. For now, all IOS platforms
instantiate VideoPresentationInterfaceAVKit.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/platform/graphics/PlatformVideoPresentationInterface.h:
* Source/WebCore/platform/ios/VideoPresentationInterfaceAVKit.h: Added.
(WebCore::VideoPresentationInterfaceAVKit::logClassName const):
(WebCore::VideoPresentationInterfaceAVKit::playerLayerView const):
(WebCore::VideoPresentationInterfaceAVKit::changingStandbyOnly):
(WebCore::VideoPresentationInterfaceAVKit::allowsPictureInPicturePlayback const):
* Source/WebCore/platform/ios/VideoPresentationInterfaceAVKit.mm: Copied from Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.mm.
(clearUIColor):
(boolString):
(-[WebAVPlayerViewControllerDelegate fullscreenInterface]):
(-[WebAVPlayerViewControllerDelegate setFullscreenInterface:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewController:failedToStartPictureInPictureWithError:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldAutomaticallyDismissAtPictureInPictureStart:]):
(convertToExitFullScreenReason):
(-[WebAVPlayerViewControllerDelegate playerViewController:shouldExitFullScreenWithReason:]):
(-[WebAVPlayerViewControllerDelegate playerViewController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldStartPictureInPictureFromInlineWhenEnteringBackground:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:failedToStartPictureInPictureWithError:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]):
(WebAVPictureInPictureContentViewController_initWithController):
(WebAVPictureInPictureContentViewController_controller):
(WebAVPictureInPictureContentViewController_playerLayer):
(WebAVPictureInPictureContentViewController_setPlayerLayer):
(WebAVPictureInPictureContentViewController_viewWillLayoutSubviews):
(WebAVPictureInPictureContentViewController_dealloc):
(allocWebAVPictureInPictureContentViewControllerInstance):
(-[WebAVPlayerViewController initWithFullscreenInterface:]):
(-[WebAVPlayerViewController configurePlayerViewControllerWithFullscreenInterface:]):
(-[WebAVPlayerViewController dealloc]):
(-[WebAVPlayerViewController playerViewControllerShouldHandleDoneButtonTap:]):
(-[WebAVPlayerViewController setWebKitOverrideRouteSharingPolicy:routingContextUID:]):
(-[WebAVPlayerViewController enterFullScreenAnimated:completionHandler:]):
(-[WebAVPlayerViewController exitFullScreenAnimated:completionHandler:]):
(-[WebAVPlayerViewController observeValueForKeyPath:ofObject:change:context:]):
(-[WebAVPlayerViewController initObserver]):
(-[WebAVPlayerViewController removeObserver]):
(-[WebAVPlayerViewController MY_NO_RETURN]):
(-[WebAVPlayerViewController isPictureInPicturePossible]):
(-[WebAVPlayerViewController isPictureInPictureActive]):
(-[WebAVPlayerViewController pictureInPictureActive]):
(-[WebAVPlayerViewController pictureInPictureWasStartedWhenEnteringBackground]):
(-[WebAVPlayerViewController view]):
(-[WebAVPlayerViewController flashPlaybackControlsWithDuration:]):
(-[WebAVPlayerViewController showsPlaybackControls]):
(-[WebAVPlayerViewController setShowsPlaybackControls:]):
(-[WebAVPlayerViewController setAllowsPictureInPicturePlayback:]):
(-[WebAVPlayerViewController setDelegate:]):
(-[WebAVPlayerViewController setPlayerController:]):
(-[WebAVPlayerViewController avPlayerViewController]):
(-[WebAVPlayerViewController removeFromParentViewController]):
(-[WebAVPlayerViewController logIdentifier]):
(-[WebAVPlayerViewController loggerPtr]):
(-[WebAVPlayerViewController logChannel]):
(VideoPresentationInterfaceAVKit::create):
(VideoPresentationInterfaceAVKit::VideoPresentationInterfaceAVKit):
(VideoPresentationInterfaceAVKit::~VideoPresentationInterfaceAVKit):
(VideoPresentationInterfaceAVKit::avPlayerViewController const):
(VideoPresentationInterfaceAVKit::videoDimensionsChanged):
(VideoPresentationInterfaceAVKit::setupFullscreen):
(VideoPresentationInterfaceAVKit::setVideoPresentationModel):
(VideoPresentationInterfaceAVKit::hasVideoChanged):
(VideoPresentationInterfaceAVKit::externalPlaybackChanged):
(VideoPresentationInterfaceAVKit::pictureInPictureWasStartedWhenEnteringBackground const):
(VideoPresentationInterfaceAVKit::exitFullscreen):
(VideoPresentationInterfaceAVKit::cleanupFullscreen):
(VideoPresentationInterfaceAVKit::setPlayerIdentifier):
(VideoPresentationInterfaceAVKit::requestHideAndExitFullscreen):
(VideoPresentationInterfaceAVKit::preparedToReturnToInline):
(VideoPresentationInterfaceAVKit::mayAutomaticallyShowVideoPictureInPicture const):
(VideoPresentationInterfaceAVKit::willStartPictureInPicture):
(VideoPresentationInterfaceAVKit::didStartPictureInPicture):
(VideoPresentationInterfaceAVKit::failedToStartPictureInPicture):
(VideoPresentationInterfaceAVKit::didStopPictureInPicture):
(VideoPresentationInterfaceAVKit::prepareForPictureInPictureStopWithCompletionHandler):
(VideoPresentationInterfaceAVKit::shouldExitFullscreenWithReason):
(VideoPresentationInterfaceAVKit::setInlineRect):
(VideoPresentationInterfaceAVKit::doSetup):
(VideoPresentationInterfaceAVKit::doEnterFullscreen):
(VideoPresentationInterfaceAVKit::doExitFullscreen):
(VideoPresentationInterfaceAVKit::exitFullscreenHandler):
(VideoPresentationInterfaceAVKit::enterFullscreenHandler):
(VideoPresentationInterfaceAVKit::watchdogTimerFired):
(VideoPresentationInterfaceAVKit::isPlayingVideoInEnhancedFullscreen const):
(WebCore::setSupportsPictureInPicture):
(WebCore::supportsPictureInPicture):
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.h:
(WebCore::VideoPresentationInterfaceIOS::hasMode const):
(WebCore::VideoPresentationInterfaceIOS::playerIdentifier const):
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.mm:
(VideoPresentationInterfaceIOS::VideoPresentationInterfaceIOS):
(VideoPresentationInterfaceIOS::~VideoPresentationInterfaceIOS):
(VideoPresentationInterfaceIOS::playerController const):
(VideoPresentationInterfaceIOS::applicationDidBecomeActive):
(clearUIColor): Deleted.
(boolString): Deleted.
(-[WebAVPlayerViewControllerDelegate fullscreenInterface]): Deleted.
(-[WebAVPlayerViewControllerDelegate setFullscreenInterface:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewController:failedToStartPictureInPictureWithError:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldAutomaticallyDismissAtPictureInPictureStart:]): Deleted.
(convertToExitFullScreenReason): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewController:shouldExitFullScreenWithReason:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldDismiss:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldStartPictureInPictureFromInlineWhenEnteringBackground:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:failedToStartPictureInPictureWithError:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]): Deleted.
(WebAVPictureInPictureContentViewController_initWithController): Deleted.
(WebAVPictureInPictureContentViewController_controller): Deleted.
(WebAVPictureInPictureContentViewController_playerLayer): Deleted.
(WebAVPictureInPictureContentViewController_setPlayerLayer): Deleted.
(WebAVPictureInPictureContentViewController_viewWillLayoutSubviews): Deleted.
(WebAVPictureInPictureContentViewController_dealloc): Deleted.
(allocWebAVPictureInPictureContentViewControllerInstance): Deleted.
(-[WebAVPlayerViewController initWithFullscreenInterface:]): Deleted.
(-[WebAVPlayerViewController configurePlayerViewControllerWithFullscreenInterface:]): Deleted.
(-[WebAVPlayerViewController dealloc]): Deleted.
(-[WebAVPlayerViewController playerViewControllerShouldHandleDoneButtonTap:]): Deleted.
(-[WebAVPlayerViewController setWebKitOverrideRouteSharingPolicy:routingContextUID:]): Deleted.
(-[WebAVPlayerViewController enterFullScreenAnimated:completionHandler:]): Deleted.
(-[WebAVPlayerViewController exitFullScreenAnimated:completionHandler:]): Deleted.
(-[WebAVPlayerViewController observeValueForKeyPath:ofObject:change:context:]): Deleted.
(-[WebAVPlayerViewController initObserver]): Deleted.
(-[WebAVPlayerViewController removeObserver]): Deleted.
(-[WebAVPlayerViewController MY_NO_RETURN]): Deleted.
(-[WebAVPlayerViewController isPictureInPicturePossible]): Deleted.
(-[WebAVPlayerViewController isPictureInPictureActive]): Deleted.
(-[WebAVPlayerViewController pictureInPictureActive]): Deleted.
(-[WebAVPlayerViewController pictureInPictureWasStartedWhenEnteringBackground]): Deleted.
(-[WebAVPlayerViewController view]): Deleted.
(-[WebAVPlayerViewController flashPlaybackControlsWithDuration:]): Deleted.
(-[WebAVPlayerViewController showsPlaybackControls]): Deleted.
(-[WebAVPlayerViewController setShowsPlaybackControls:]): Deleted.
(-[WebAVPlayerViewController setAllowsPictureInPicturePlayback:]): Deleted.
(-[WebAVPlayerViewController setDelegate:]): Deleted.
(-[WebAVPlayerViewController setPlayerController:]): Deleted.
(-[WebAVPlayerViewController avPlayerViewController]): Deleted.
(-[WebAVPlayerViewController removeFromParentViewController]): Deleted.
(-[WebAVPlayerViewController logIdentifier]): Deleted.
(-[WebAVPlayerViewController loggerPtr]): Deleted.
(-[WebAVPlayerViewController logChannel]): Deleted.
(VideoPresentationInterfaceIOS::create): Deleted.
(VideoPresentationInterfaceIOS::avPlayerViewController const): Deleted.
(VideoPresentationInterfaceIOS::setVideoPresentationModel): Deleted.
(VideoPresentationInterfaceIOS::hasVideoChanged): Deleted.
(VideoPresentationInterfaceIOS::videoDimensionsChanged): Deleted.
(VideoPresentationInterfaceIOS::externalPlaybackChanged): Deleted.
(VideoPresentationInterfaceIOS::pictureInPictureWasStartedWhenEnteringBackground const): Deleted.
(VideoPresentationInterfaceIOS::setupFullscreen): Deleted.
(VideoPresentationInterfaceIOS::exitFullscreen): Deleted.
(VideoPresentationInterfaceIOS::cleanupFullscreen): Deleted.
(VideoPresentationInterfaceIOS::setPlayerIdentifier): Deleted.
(VideoPresentationInterfaceIOS::requestHideAndExitFullscreen): Deleted.
(VideoPresentationInterfaceIOS::preparedToReturnToInline): Deleted.
(VideoPresentationInterfaceIOS::mayAutomaticallyShowVideoPictureInPicture const): Deleted.
(VideoPresentationInterfaceIOS::willStartPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::didStartPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::failedToStartPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::didStopPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::prepareForPictureInPictureStopWithCompletionHandler): Deleted.
(VideoPresentationInterfaceIOS::shouldExitFullscreenWithReason): Deleted.
(VideoPresentationInterfaceIOS::setInlineRect): Deleted.
(VideoPresentationInterfaceIOS::doSetup): Deleted.
(VideoPresentationInterfaceIOS::doEnterFullscreen): Deleted.
(VideoPresentationInterfaceIOS::doExitFullscreen): Deleted.
(VideoPresentationInterfaceIOS::exitFullscreenHandler): Deleted.
(VideoPresentationInterfaceIOS::enterFullscreenHandler): Deleted.
(VideoPresentationInterfaceIOS::watchdogTimerFired): Deleted.
(VideoPresentationInterfaceIOS::isPlayingVideoInEnhancedFullscreen const): Deleted.
(WebCore::setSupportsPictureInPicture): Deleted.
(WebCore::supportsPictureInPicture): Deleted.
* Source/WebCore/platform/ios/VideoPresentationInterfaceLMK.h: Added.
(WebCore::VideoPresentationInterfaceLMK::logClassName const):
* Source/WebCore/platform/ios/VideoPresentationInterfaceLMK.mm: Added.
(VideoPresentationInterfaceLMK::~VideoPresentationInterfaceLMK):
(VideoPresentationInterfaceLMK::create):
(VideoPresentationInterfaceLMK::VideoPresentationInterfaceLMK):
(VideoPresentationInterfaceLMK::setupFullscreen):
(VideoPresentationInterfaceLMK::setVideoPresentationModel):
(VideoPresentationInterfaceLMK::videoDimensionsChanged):
(VideoPresentationInterfaceLMK::exitFullscreen):
(VideoPresentationInterfaceLMK::pictureInPictureWasStartedWhenEnteringBackground const):
(VideoPresentationInterfaceLMK::hasVideoChanged):
(VideoPresentationInterfaceLMK::externalPlaybackChanged):
(VideoPresentationInterfaceLMK::cleanupFullscreen):
(VideoPresentationInterfaceLMK::setPlayerIdentifier):
(VideoPresentationInterfaceLMK::requestHideAndExitFullscreen):
(VideoPresentationInterfaceLMK::preparedToReturnToInline):
(VideoPresentationInterfaceLMK::mayAutomaticallyShowVideoPictureInPicture const):
(VideoPresentationInterfaceLMK::willStartPictureInPicture):
(VideoPresentationInterfaceLMK::didStartPictureInPicture):
(VideoPresentationInterfaceLMK::failedToStartPictureInPicture):
(VideoPresentationInterfaceLMK::didStopPictureInPicture):
(VideoPresentationInterfaceLMK::prepareForPictureInPictureStopWithCompletionHandler):
(VideoPresentationInterfaceLMK::shouldExitFullscreenWithReason):
(VideoPresentationInterfaceLMK::setInlineRect):
(VideoPresentationInterfaceLMK::doSetup):
(VideoPresentationInterfaceLMK::doEnterFullscreen):
(VideoPresentationInterfaceLMK::doExitFullscreen):
(VideoPresentationInterfaceLMK::exitFullscreenHandler):
(VideoPresentationInterfaceLMK::enterFullscreenHandler):
(VideoPresentationInterfaceLMK::watchdogTimerFired):
(VideoPresentationInterfaceLMK::isPlayingVideoInEnhancedFullscreen const):
(VideoPresentationInterfaceLMK::avPlayerViewController const):
(VideoPresentationInterfaceLMK::playerLayerView const):
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
(VideoFullscreenControllerContext::setUpFullscreen):
* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:

Canonical link: <a href="https://commits.webkit.org/274159@main">https://commits.webkit.org/274159@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a538504ac74dd603cf638ad97c0cc92d11597496

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37637 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16516 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39882 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40167 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33504 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38954 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19246 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13678 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31882 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38202 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13962 "Found 2 new test failures: fast/dynamic/anchor-lock.html, media/video-ended-does-not-hold-sleep-assertion.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32964 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12138 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12131 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33647 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41428 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34086 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34069 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37976 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12667 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10201 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36144 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14103 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8545 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13067 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13416 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/platform/ios/VideoPresentationInterfaceLMK.mm:
```diff
@@ -0,0 +1,205 @@
+}
+
+void VideoPresentationInterfaceLMK::setVideoPresentationModel(VideoPresentationModel* model)
+{
```
you can just omit the parameter names!

jernoble commented on Source/WebCore/platform/ios/VideoPresentationInterfaceAVKit.mm:
Looks like this is causing a build error:
```
/Volumes/Data/worker/watchOS-10-Build-EWS/build/Source/WebCore/platform/ios/VideoPresentationInterfaceAVKit.mm:1171:5: error: 'm_waitingForPreparedToExit' is a private member of 'WebCore::VideoPresentationInterfaceIOS'
```

We should move that ivar to `protected:` or add a `bool waitingForPreparedToExit() const`/`void setWaitingForPreparedToExit(bool)` to the IOS base class.

danae404 replied:
Good catch- thanks

aestes commented on Source/WebCore/platform/ios/VideoPresentationInterfaceAVKit.h:
```diff
@@ -0,0 +1,99 @@
+
+#if PLATFORM(IOS_FAMILY) && HAVE(AVKIT)
+
+#include "PlaybackSessionInterfaceAVKit.h"
```
Can you just forward-declare `class PlaybackSessionInterfaceAVKit` instead of including its header?

danae404 committed:
Refactor VideoPresentationInterfaceIOS
https://bugs.webkit.org/show_bug.cgi?id=268636
rdar://122193331

Reviewed by Andy Estes.

This patch extracts the AVKit-specific code from VideoPresentationInterfaceIOS
and places it into a subclass called VideoPresentationInterfaceAVKit. There
is also now a second subclass called VideoPresentationInterfaceLMK, which
will be used on the vision platform in the future. For now, all IOS platforms
instantiate VideoPresentationInterfaceAVKit.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/platform/graphics/PlatformVideoPresentationInterface.h:
* Source/WebCore/platform/ios/VideoPresentationInterfaceAVKit.h: Added.
(WebCore::VideoPresentationInterfaceAVKit::logClassName const):
(WebCore::VideoPresentationInterfaceAVKit::playerLayerView const):
(WebCore::VideoPresentationInterfaceAVKit::changingStandbyOnly):
(WebCore::VideoPresentationInterfaceAVKit::allowsPictureInPicturePlayback const):
* Source/WebCore/platform/ios/VideoPresentationInterfaceAVKit.mm: Copied from Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.mm.
(clearUIColor):
(boolString):
(-[WebAVPlayerViewControllerDelegate fullscreenInterface]):
(-[WebAVPlayerViewControllerDelegate setFullscreenInterface:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewController:failedToStartPictureInPictureWithError:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldAutomaticallyDismissAtPictureInPictureStart:]):
(convertToExitFullScreenReason):
(-[WebAVPlayerViewControllerDelegate playerViewController:shouldExitFullScreenWithReason:]):
(-[WebAVPlayerViewControllerDelegate playerViewController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]):
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldStartPictureInPictureFromInlineWhenEnteringBackground:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStartPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:failedToStartPictureInPictureWithError:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStopPictureInPicture:]):
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]):
(WebAVPictureInPictureContentViewController_initWithController):
(WebAVPictureInPictureContentViewController_controller):
(WebAVPictureInPictureContentViewController_playerLayer):
(WebAVPictureInPictureContentViewController_setPlayerLayer):
(WebAVPictureInPictureContentViewController_viewWillLayoutSubviews):
(WebAVPictureInPictureContentViewController_dealloc):
(allocWebAVPictureInPictureContentViewControllerInstance):
(-[WebAVPlayerViewController initWithFullscreenInterface:]):
(-[WebAVPlayerViewController configurePlayerViewControllerWithFullscreenInterface:]):
(-[WebAVPlayerViewController dealloc]):
(-[WebAVPlayerViewController playerViewControllerShouldHandleDoneButtonTap:]):
(-[WebAVPlayerViewController setWebKitOverrideRouteSharingPolicy:routingContextUID:]):
(-[WebAVPlayerViewController enterFullScreenAnimated:completionHandler:]):
(-[WebAVPlayerViewController exitFullScreenAnimated:completionHandler:]):
(-[WebAVPlayerViewController observeValueForKeyPath:ofObject:change:context:]):
(-[WebAVPlayerViewController initObserver]):
(-[WebAVPlayerViewController removeObserver]):
(-[WebAVPlayerViewController MY_NO_RETURN]):
(-[WebAVPlayerViewController isPictureInPicturePossible]):
(-[WebAVPlayerViewController isPictureInPictureActive]):
(-[WebAVPlayerViewController pictureInPictureActive]):
(-[WebAVPlayerViewController pictureInPictureWasStartedWhenEnteringBackground]):
(-[WebAVPlayerViewController view]):
(-[WebAVPlayerViewController flashPlaybackControlsWithDuration:]):
(-[WebAVPlayerViewController showsPlaybackControls]):
(-[WebAVPlayerViewController setShowsPlaybackControls:]):
(-[WebAVPlayerViewController setAllowsPictureInPicturePlayback:]):
(-[WebAVPlayerViewController setDelegate:]):
(-[WebAVPlayerViewController setPlayerController:]):
(-[WebAVPlayerViewController avPlayerViewController]):
(-[WebAVPlayerViewController removeFromParentViewController]):
(-[WebAVPlayerViewController logIdentifier]):
(-[WebAVPlayerViewController loggerPtr]):
(-[WebAVPlayerViewController logChannel]):
(VideoPresentationInterfaceAVKit::create):
(VideoPresentationInterfaceAVKit::VideoPresentationInterfaceAVKit):
(VideoPresentationInterfaceAVKit::~VideoPresentationInterfaceAVKit):
(VideoPresentationInterfaceAVKit::avPlayerViewController const):
(VideoPresentationInterfaceAVKit::videoDimensionsChanged):
(VideoPresentationInterfaceAVKit::setupFullscreen):
(VideoPresentationInterfaceAVKit::setVideoPresentationModel):
(VideoPresentationInterfaceAVKit::hasVideoChanged):
(VideoPresentationInterfaceAVKit::externalPlaybackChanged):
(VideoPresentationInterfaceAVKit::pictureInPictureWasStartedWhenEnteringBackground const):
(VideoPresentationInterfaceAVKit::exitFullscreen):
(VideoPresentationInterfaceAVKit::cleanupFullscreen):
(VideoPresentationInterfaceAVKit::setPlayerIdentifier):
(VideoPresentationInterfaceAVKit::requestHideAndExitFullscreen):
(VideoPresentationInterfaceAVKit::preparedToReturnToInline):
(VideoPresentationInterfaceAVKit::mayAutomaticallyShowVideoPictureInPicture const):
(VideoPresentationInterfaceAVKit::willStartPictureInPicture):
(VideoPresentationInterfaceAVKit::didStartPictureInPicture):
(VideoPresentationInterfaceAVKit::failedToStartPictureInPicture):
(VideoPresentationInterfaceAVKit::didStopPictureInPicture):
(VideoPresentationInterfaceAVKit::prepareForPictureInPictureStopWithCompletionHandler):
(VideoPresentationInterfaceAVKit::shouldExitFullscreenWithReason):
(VideoPresentationInterfaceAVKit::setInlineRect):
(VideoPresentationInterfaceAVKit::doSetup):
(VideoPresentationInterfaceAVKit::doEnterFullscreen):
(VideoPresentationInterfaceAVKit::doExitFullscreen):
(VideoPresentationInterfaceAVKit::exitFullscreenHandler):
(VideoPresentationInterfaceAVKit::enterFullscreenHandler):
(VideoPresentationInterfaceAVKit::watchdogTimerFired):
(VideoPresentationInterfaceAVKit::isPlayingVideoInEnhancedFullscreen const):
(WebCore::setSupportsPictureInPicture):
(WebCore::supportsPictureInPicture):
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.h:
(WebCore::VideoPresentationInterfaceIOS::hasMode const):
(WebCore::VideoPresentationInterfaceIOS::playerIdentifier const):
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.mm:
(VideoPresentationInterfaceIOS::VideoPresentationInterfaceIOS):
(VideoPresentationInterfaceIOS::~VideoPresentationInterfaceIOS):
(VideoPresentationInterfaceIOS::playerController const):
(VideoPresentationInterfaceIOS::applicationDidBecomeActive):
(clearUIColor): Deleted.
(boolString): Deleted.
(-[WebAVPlayerViewControllerDelegate fullscreenInterface]): Deleted.
(-[WebAVPlayerViewControllerDelegate setFullscreenInterface:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewController:failedToStartPictureInPictureWithError:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerWillStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerDidStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldAutomaticallyDismissAtPictureInPictureStart:]): Deleted.
(convertToExitFullScreenReason): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewController:shouldExitFullScreenWithReason:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldDismiss:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]): Deleted.
(-[WebAVPlayerViewControllerDelegate playerViewControllerShouldStartPictureInPictureFromInlineWhenEnteringBackground:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStartPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:failedToStartPictureInPictureWithError:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerWillStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureControllerDidStopPictureInPicture:]): Deleted.
(-[WebAVPlayerViewControllerDelegate pictureInPictureController:restoreUserInterfaceForPictureInPictureStopWithCompletionHandler:]): Deleted.
(WebAVPictureInPictureContentViewController_initWithController): Deleted.
(WebAVPictureInPictureContentViewController_controller): Deleted.
(WebAVPictureInPictureContentViewController_playerLayer): Deleted.
(WebAVPictureInPictureContentViewController_setPlayerLayer): Deleted.
(WebAVPictureInPictureContentViewController_viewWillLayoutSubviews): Deleted.
(WebAVPictureInPictureContentViewController_dealloc): Deleted.
(allocWebAVPictureInPictureContentViewControllerInstance): Deleted.
(-[WebAVPlayerViewController initWithFullscreenInterface:]): Deleted.
(-[WebAVPlayerViewController configurePlayerViewControllerWithFullscreenInterface:]): Deleted.
(-[WebAVPlayerViewController dealloc]): Deleted.
(-[WebAVPlayerViewController playerViewControllerShouldHandleDoneButtonTap:]): Deleted.
(-[WebAVPlayerViewController setWebKitOverrideRouteSharingPolicy:routingContextUID:]): Deleted.
(-[WebAVPlayerViewController enterFullScreenAnimated:completionHandler:]): Deleted.
(-[WebAVPlayerViewController exitFullScreenAnimated:completionHandler:]): Deleted.
(-[WebAVPlayerViewController observeValueForKeyPath:ofObject:change:context:]): Deleted.
(-[WebAVPlayerViewController initObserver]): Deleted.
(-[WebAVPlayerViewController removeObserver]): Deleted.
(-[WebAVPlayerViewController MY_NO_RETURN]): Deleted.
(-[WebAVPlayerViewController isPictureInPicturePossible]): Deleted.
(-[WebAVPlayerViewController isPictureInPictureActive]): Deleted.
(-[WebAVPlayerViewController pictureInPictureActive]): Deleted.
(-[WebAVPlayerViewController pictureInPictureWasStartedWhenEnteringBackground]): Deleted.
(-[WebAVPlayerViewController view]): Deleted.
(-[WebAVPlayerViewController flashPlaybackControlsWithDuration:]): Deleted.
(-[WebAVPlayerViewController showsPlaybackControls]): Deleted.
(-[WebAVPlayerViewController setShowsPlaybackControls:]): Deleted.
(-[WebAVPlayerViewController setAllowsPictureInPicturePlayback:]): Deleted.
(-[WebAVPlayerViewController setDelegate:]): Deleted.
(-[WebAVPlayerViewController setPlayerController:]): Deleted.
(-[WebAVPlayerViewController avPlayerViewController]): Deleted.
(-[WebAVPlayerViewController removeFromParentViewController]): Deleted.
(-[WebAVPlayerViewController logIdentifier]): Deleted.
(-[WebAVPlayerViewController loggerPtr]): Deleted.
(-[WebAVPlayerViewController logChannel]): Deleted.
(VideoPresentationInterfaceIOS::create): Deleted.
(VideoPresentationInterfaceIOS::avPlayerViewController const): Deleted.
(VideoPresentationInterfaceIOS::setVideoPresentationModel): Deleted.
(VideoPresentationInterfaceIOS::hasVideoChanged): Deleted.
(VideoPresentationInterfaceIOS::videoDimensionsChanged): Deleted.
(VideoPresentationInterfaceIOS::externalPlaybackChanged): Deleted.
(VideoPresentationInterfaceIOS::pictureInPictureWasStartedWhenEnteringBackground const): Deleted.
(VideoPresentationInterfaceIOS::setupFullscreen): Deleted.
(VideoPresentationInterfaceIOS::exitFullscreen): Deleted.
(VideoPresentationInterfaceIOS::cleanupFullscreen): Deleted.
(VideoPresentationInterfaceIOS::setPlayerIdentifier): Deleted.
(VideoPresentationInterfaceIOS::requestHideAndExitFullscreen): Deleted.
(VideoPresentationInterfaceIOS::preparedToReturnToInline): Deleted.
(VideoPresentationInterfaceIOS::mayAutomaticallyShowVideoPictureInPicture const): Deleted.
(VideoPresentationInterfaceIOS::willStartPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::didStartPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::failedToStartPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::didStopPictureInPicture): Deleted.
(VideoPresentationInterfaceIOS::prepareForPictureInPictureStopWithCompletionHandler): Deleted.
(VideoPresentationInterfaceIOS::shouldExitFullscreenWithReason): Deleted.
(VideoPresentationInterfaceIOS::setInlineRect): Deleted.
(VideoPresentationInterfaceIOS::doSetup): Deleted.
(VideoPresentationInterfaceIOS::doEnterFullscreen): Deleted.
(VideoPresentationInterfaceIOS::doExitFullscreen): Deleted.
(VideoPresentationInterfaceIOS::exitFullscreenHandler): Deleted.
(VideoPresentationInterfaceIOS::enterFullscreenHandler): Deleted.
(VideoPresentationInterfaceIOS::watchdogTimerFired): Deleted.
(VideoPresentationInterfaceIOS::isPlayingVideoInEnhancedFullscreen const): Deleted.
(WebCore::setSupportsPictureInPicture): Deleted.
(WebCore::supportsPictureInPicture): Deleted.
* Source/WebCore/platform/ios/VideoPresentationInterfaceLMK.h: Added.
(WebCore::VideoPresentationInterfaceLMK::logClassName const):
* Source/WebCore/platform/ios/VideoPresentationInterfaceLMK.mm: Added.
(VideoPresentationInterfaceLMK::~VideoPresentationInterfaceLMK):
(VideoPresentationInterfaceLMK::create):
(VideoPresentationInterfaceLMK::VideoPresentationInterfaceLMK):
(VideoPresentationInterfaceLMK::setupFullscreen):
(VideoPresentationInterfaceLMK::setVideoPresentationModel):
(VideoPresentationInterfaceLMK::videoDimensionsChanged):
(VideoPresentationInterfaceLMK::exitFullscreen):
(VideoPresentationInterfaceLMK::pictureInPictureWasStartedWhenEnteringBackground const):
(VideoPresentationInterfaceLMK::hasVideoChanged):
(VideoPresentationInterfaceLMK::externalPlaybackChanged):
(VideoPresentationInterfaceLMK::cleanupFullscreen):
(VideoPresentationInterfaceLMK::setPlayerIdentifier):
(VideoPresentationInterfaceLMK::requestHideAndExitFullscreen):
(VideoPresentationInterfaceLMK::preparedToReturnToInline):
(VideoPresentationInterfaceLMK::mayAutomaticallyShowVideoPictureInPicture const):
(VideoPresentationInterfaceLMK::willStartPictureInPicture):
(VideoPresentationInterfaceLMK::didStartPictureInPicture):
(VideoPresentationInterfaceLMK::failedToStartPictureInPicture):
(VideoPresentationInterfaceLMK::didStopPictureInPicture):
(VideoPresentationInterfaceLMK::prepareForPictureInPictureStopWithCompletionHandler):
(VideoPresentationInterfaceLMK::shouldExitFullscreenWithReason):
(VideoPresentationInterfaceLMK::setInlineRect):
(VideoPresentationInterfaceLMK::doSetup):
(VideoPresentationInterfaceLMK::doEnterFullscreen):
(VideoPresentationInterfaceLMK::doExitFullscreen):
(VideoPresentationInterfaceLMK::exitFullscreenHandler):
(VideoPresentationInterfaceLMK::enterFullscreenHandler):
(VideoPresentationInterfaceLMK::watchdogTimerFired):
(VideoPresentationInterfaceLMK::isPlayingVideoInEnhancedFullscreen const):
(VideoPresentationInterfaceLMK::avPlayerViewController const):
(VideoPresentationInterfaceLMK::playerLayerView const):
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
(VideoFullscreenControllerContext::setUpFullscreen):
* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:

Canonical link: https://commits.webkit.org/274159@main
